### ISSUE TITLE
Parameter of set literal containing tuples of disparate values is unknown

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1722]
**Describe the bug**
The type parameter of a set literal containing tuples of disparate values is unknown.  The curly brackets are also removed from the printed message.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That the type of the first set would be `set[tuple[str]]]` or the `Literal` equivalent.

**Screenshots or Code**

```python
# pyright: strict

# Type of "(('a', ), ('b', ))" is "tuple[tuple[Literal['a']], tuple[Literal['b']]]"
reveal_type((
    ('a',),
    ('b',),
))

# Type of "[('a', ), ('b', )]" is "list[tuple[Literal['a']] | tuple[Literal['b']]]"
reveal_type([
    ('a',),
    ('b',),
])

# Type of "('a', ), ('b', )" is "set[Unknown]"
reveal_type({
    ('a',),
    ('b',),
})

# Type of "set(('a', ), ('b', ))" is "set[tuple[Literal['a']] | tuple[Literal['b']]]"
reveal_type(set({
    ('a',),
    ('b',),
}))

```

**VS Code extension or command-line**
1.1.128

**Comment by erictraut** [comment]
This is due to the fact that pyright supports a `strictListInference` and `strictDictionaryInference` configuration setting, but there is no equivalent `strictSetInference`. I addressed that inconsistency and added a `strictSetInference` configuration setting. It is enabled by default in "strict" mode.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.129, which I just published. It will also be included in the next release of pylance.
